docs/adr/0004-support-hot-migration.md

```diff
@@ -0,0 +1,61 @@
+# 4. Support Hot Migration
+
+* Status: proposed
+* Deciders: @butonic, @micbar, @dragotin, @hodyroff, @pmaier1
+* Date: 2021-03-16
+
+Technical Story: \[description | ticket/issue URL\]
+
+## Context and Problem Statement
+
+Migration is one of the most important topics of the oCIS story. We need to provide a concept how to migrate from oC10 to oCIS.
+
+## Decision Drivers
+
+- Do not lose file blob or meta data.
+- Existing shares like public links and federated shares must remain functional after the migration.
+- Legacy clients need to keep working
```

Comment:
I think we need to be a bit more accurate here: Which meta data do we need to keep? Which share info do we need to preserve exactly? Which API's for old clients do we still support?